### PR TITLE
Filterable map views

### PR DESCRIPTION
**Pull request by Twey**
## Motivation

We have queries in our GraphQL schema that are parameterized by scalars such as `ChannelFullName` or `Target`.  These scalars, however, are difficult to guess, which makes exploring these aspects of the node difficult.

## Proposal

Augment the existing singular `key` → `value` representation of maps (or specifically `CollectionView`, since that's the type used for these structures) with a plural `inputs` → `values` query.  `inputs` is optional, and leaving it out allows retrieving all values in the map without needing to know the keys; it also currently supports selecting a subset of known keys by the `filters.keys` property, as a primitive filter.

In the future, additional fields can be added to `inputs`, allowing for more sophisticated predicate filtering as well as things like pagination.

## Test Plan

I think that CI ensures that this API breakage doesn't bring down anything critical, but to experience the discoverability, enter GraphiQL and try completing e.g. `chain(chainId: $chainId) { inboxes { values { origin } } }`.  `$chainId` must unfortunately currently be retrieved with a separate query, e.g. `chains { default }`.

## Release Plan

This PR breaks GraphQL API compatibility.  I'm not sure how this fits into our versioning system.

- Need to bump the major/minor version number in the next release of the crates.
- Release is blocked and/or tracked by other issues (see links below)

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)
- fixes https://github.com/linera-io/linera-protocol/issues/1199

### CODE REVIEW

**Comment by jvff** [comment]
Not sure if it helps, but I used to have OOM issues (with 64GiB RAM) when too many things were running in paralllel (especially linking test binaries or executing some of the tests). So one think that helped (before I eventually upgraded to have more memory 😅) was to limit parallelism using the `-j` flag with Cargo.

File: linera-views-derive/src/snapshots/linera_views_derive__tests__generate_crypto_hash_code-2.snap [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+source: linera-views-derive/src/lib.rs
+assertion_line: 793
+expression: generate_crypto_hash_code(input)
+---
+# [async_trait :: async_trait] impl linera_views :: views :: CryptoHashView < CustomContext > for TestView { async fn crypto_hash (& self) -> Result < linera_base :: crypto :: CryptoHash , linera_views :: views :: ViewError > { use linera_base :: crypto :: { BcsHashable , CryptoHash } ; use linera_views :: { batch :: Batch , generic_array :: GenericArray , sha3 :: { digest :: OutputSizeUser , Sha3_256 } , views :: HashableView , } ; use serde :: { Serialize , Deserialize } ; # [derive (Serialize , Deserialize)] struct TestViewHash (GenericArray < u8 , < Sha3_256 as OutputSizeUser > :: OutputSize >) ; impl BcsHashable for TestViewHash { } let hash = self . hash () . await ? ; Ok (CryptoHash :: new (& TestViewHash (hash))) } }
```
Comment (ma2bd) [inline linera-views-derive/src/snapshots/linera_views_derive__tests__generate_crypto_hash_code-2.snap:6]: Any idea how to make this readable though?

**Review by ma2bd — APPROVED** [review]
Nice!

**Comment by ma2bd** [comment]
Can you fix the release plan? (Yes this is breaking change. No there are no new features)

File: linera-views-derive/src/snapshots/linera_views_derive__tests__generate_crypto_hash_code-2.snap (hunk 1 above)
Comment (Twey) [reply linera-views-derive/src/snapshots/linera_views_derive__tests__generate_crypto_hash_code-2.snap]: I've run it through [`prettyplease`](https://github.com/dtolnay/prettyplease).  That should make it a lot more readable :)

**Pull request merged**